quarkus/extensions/src/main/java/org/keycloak/provider/quarkus/QuarkusFilter.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates
+ * and other contributors as indicated by the @author tags.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.keycloak.provider.quarkus;
+
+import io.vertx.core.http.HttpServerRequest;
+import org.jboss.resteasy.core.interception.jaxrs.ContainerResponseContextImpl;
+import org.keycloak.common.ClientConnection;
+import org.keycloak.common.util.Resteasy;
+import org.keycloak.models.KeycloakSession;
+import org.keycloak.models.KeycloakSessionFactory;
+import org.keycloak.models.KeycloakTransaction;
+import org.keycloak.services.resources.KeycloakApplication;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.annotation.Priority;
+import javax.inject.Inject;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.container.ContainerResponseContext;
+import javax.ws.rs.container.PreMatching;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.NewCookie;
+import javax.ws.rs.ext.Provider;
+import java.io.IOException;
+import java.util.ArrayList;
+
+
+@PreMatching
+@Provider
+@Priority(1)
+public class QuarkusFilter implements javax.ws.rs.container.ContainerRequestFilter,
+        javax.ws.rs.container.ContainerResponseFilter  {
```

Comment:
I'm not sure whether or not a `ContainerResponseFilter` is invoked if an exception is thrown by applications. If not, we won't be able to perform the `ending` aspects of this filter such as cleaning up sessions, transactions, etc.